src/audio/kpb.c

```diff
@@ -64,6 +64,7 @@ struct comp_data {
 	bool sync_draining_mode; /**< should we synchronize draining with
 				   * host?
 				   */
+	spinlock_t lock;	/**< locking mechanism */
```

Comment:
minor, it's always good practice to name/comment the usage here. i.e. ```/* pointer update locking */``` Btw, wont we need the same for W ptr ?

Reply:
@lgirdwood, I have updated the comment to describe current use in more details. However in the future we may use this lock in other places too.

> Btw, wont we need the same for W ptr ?

Honestly I don't see any change for race condition there at the moment. There is only one task which writes to internal history buffer so there is nothing else that can interrupt or overwrite it. Even with multiple customers of KPB there will always be just one buffering task.